tests/comparisons/Migration/testCompositeConstraintsSnapshot.php

```diff
@@ -34,26 +34,31 @@ public function up()
             ])
             ->addIndex(
                 [
-                    'slug',
-                ],
-                ['unique' => true]
+                    'category_id',
+                ]
+            )
+            ->addIndex(
+                [
+                    'product_id',
+                ]
```

Comment:
I did some research during my lunch break, and constraints should create them : 

> MySQL requires indexes on foreign keys and referenced keys so that foreign key checks can be fast and not require a table scan. In the referencing table, there must be an index where the foreign key columns are listed as the first columns in the same order. __Such an index is created on the referencing table automatically if it does not exist__. This index might be silently dropped later, if you create another index that can be used to enforce the foreign key constraint. index_name, if given, is used as described previously.

But when using "ALTER TABLE", indexes needs to be created first : 

> When you add a foreign key constraint to a table using ALTER TABLE, remember to create the required indexes first.

Source : http://dev.mysql.com/doc/refman/5.5/en/create-table-foreign-keys.html

So apparently, this is expected. We did not need this in the first implementation because FK were create on table creations so indexes were created on the fly.